### PR TITLE
Fixes to make FLAVA pretraining / finetuning run

### PR DESCRIPTION
**Pull request by ankitade**
Summary:
Some fixes to make both pretraining and finetuning work:
- change adam beta type to float
- Pass in text column names from configs to the encode_batch_text since its different for different datasets
- Allow encode_batch_text to return batch or just tokenized text

Test plan:
Manual sanity checks:
python train.py config=configs/pretraining/debug.yaml training.lightning.gpus=0 training.lightning.accelerator=cpu

  | Name  | Type                | Params
----------------------------------------------
0 | model | FLAVAForPreTraining | 357 M 
----------------------------------------------
357 M     Trainable params
0         Non-trainable params
357 M     Total params
1,430.581 Total estimated model params size (MB)
Epoch 0: : 50it [03:43,  4.47s/it, loss=15.2, v_num=17, train/losses/mmm_text_loss=10.40, train/losses/mmm_image_loss=9.120, train/losses/itm_loss=0.140, train/losses/global_Epoch 0: : 100it [06:57,  4.17s/it, loss=15.2, v_num=17, train/losses/mmm_text_loss=10.40, train/losses/mmm_image_loss=9.120, train/losses/itm_loss=0.140, train/losses/globalEpoch 0: : 100it [06:57,  4.17s/it, loss=14.2, v_num=17, train/losses/mmm_text_loss=9.950, train/losses/mmm_image_loss=9.140, train/losses/itm_loss=0.465, train/losses/global_contrastive_loss=2.090, train/losses/mim_loss=9.150, train/losses/mlm_loss=9.880]

python finetune.py config=configs/finetuning/qnli.yaml model.pretrained_model_key=flava_full training.lightning.gpus=0 training.lightning.accelerator=cpu

  | Name  | Type                   | Params
-------------------------------------------------
0 | model | FLAVAForClassification | 240 M 
-------------------------------------------------
240 M     Trainable params
0         Non-trainable params
240 M     Total params
963.069   Total estimated model params size (MB)
Epoch 0:   1%|▊                                                                 | 50/3787 [10:08<12:37:52, 12.17s/it, loss=0.715, v_num=18, train/losses/classification=0.721

--------------------------
pytest test/models/test_flava.py -vv

test/models/test_flava.py::TestFLAVA::test_forward_classification PASSED                                                                                               [ 50%]
test/models/test_flava.py::TestFLAVA::test_forward_pretraining PASSED

### CODE REVIEW

File: examples/flava/data/datamodules.py [hunk 1]
```diff
@@ -141,36 +142,38 @@ def __init__(
     ):
         super().__init__()
         self.train_dataset_infos = train_infos
+        self.text_columns = text_columns
         self.val_dataset_infos = val_infos
         if self.val_dataset_infos is None:
             self.val_dataset_infos = train_infos
         self.tokenizer = tokenizer
+        if self.tokenizer is None:
+            self.tokenizer = BertTokenizer.from_pretrained(TEXT_DEFAULT_TOKENIZER)
```
Comment (apsdehal) [inline examples/flava/data/datamodules.py:151]: Usually stuff like this should happen in setup?

File: examples/flava/train.py [hunk 2]
```diff
@@ -25,10 +25,10 @@ def main():
         seed_everything(config.training.seed, workers=True)
 
     datamodules = []
-    imagenet_datamodule = ImageDataModule(
-        **build_datamodule_kwargs(config.datasets.image, config.training)
-    )
     if "image" in config.datasets.selected:
+        imagenet_datamodule = ImageDataModule(
```
Comment (apsdehal) [inline examples/flava/train.py:29]: You need this for `MultimodalEvalCallback` below. That's why it is always initialized.

**Review by apsdehal — CHANGES_REQUESTED** [review]
Thanks for the changes! Can you confirm that you have run the integration tests (you will have to enable them manually)? Other than that, a minor comment that should be fixed.

File: examples/flava/train.py (hunk 2 above)
Comment (ankitade) [reply examples/flava/train.py]: moved it back

File: examples/flava/data/datamodules.py (hunk 1 above)
Comment (ankitade) [reply examples/flava/data/datamodules.py]: yeah i was trying to reuse it in MLMDatamodule but reverted it in favor of adding a setup to the MLM DM

**Comment by ankitade** [comment]
Updated the test plan with unit test output (after commenting skip)

File: examples/flava/utils.py [hunk 3]
```diff
@@ -10,13 +10,15 @@
 
 
 def build_datamodule_kwargs(dm_config: DatasetInfo, training_config: TrainingArguments):
```
Comment (ebsmothers) [inline examples/flava/utils.py:12]: dm_config type should be TrainingSingleDatasetInfo? Otherwise datamodule_extra_kwargs field won't exist

**Review by apsdehal — APPROVED** [review]
Except Evan's comment, rest looks good to me.

File: examples/flava/utils.py (hunk 3 above)
Comment (ankitade) [reply examples/flava/utils.py]: fixed the annotation

**Review by ebsmothers — APPROVED** [review]
